src/ape/managers/project.py

```diff
@@ -42,41 +86,128 @@ def sources(self) -> List[Path]:
+
+        # NOTE: filter by checksum, etc., and compile what's needed
+        #       to bring our cached manifest up-to-date
+        needs_compiling = filter(file_needs_compiling, self.sources)
+        contract_types.update(self.compilers.compile(list(needs_compiling)))
+
+        # Update our cached contract types in our cached manifest
+        if manifest.contractTypes:
+            manifest.contractTypes.update(contract_types)
+        else:
+            manifest.contractTypes = contract_types
+
+        # Update our cached source code entries in our cached manifest
+        cached_sources = {
+            str(source): Source(  # type: ignore
+                checksum=Checksum(algorithm="md5", hash=compute_checksum(source.read_text())),
+                urls=[],
+            )
+            for source in self.sources
+        }
+        if manifest.sources:
+            manifest.sources.update(cached_sources)
+        else:
+            manifest.sources = cached_sources
+
+        # NOTE: Cache the updated manifest to disk (so `self.cached_manifest` reads next time)
+        self.manifest_cachefile.write_text(json.dumps(manifest.to_dict()))
+
+        return contract_types
+
     @property
     def contracts(self) -> Dict[str, ContractType]:
-        return self.compilers.compile(self.sources)
+        return self._load_contracts()
+
+    def __getattr__(self, attr_name: str):
+        contracts = self._load_contracts()
+        if attr_name in contracts:
+            return contracts[attr_name]
+        elif attr_name in self.dependencies:
+            return self.dependencies[attr_name]
+        else:
+            raise AttributeError(f"{self.__class__.__name__} has no attribute '{attr_name}'")
 
+    @property
     def _interfaces_folder(self) -> Path:
         return self.path / "interfaces"
 
+    @property
     def _scripts_folder(self) -> Path:
         return self.path / "scripts"
 
+    @property
     def _tests_folder(self) -> Path:
         return self.path / "tests"
 
     # TODO: Make this work for generating and caching the manifest file
+    @property
+    def compiler_data(self) -> List[Compiler]:
+        compilers = []
+
+        for compiler in self.compilers.registered_compilers.values():
+            for version in compiler.versions:
+                compilers.append(Compiler(compiler.name, version))  # type: ignore
+
+        return compilers
+
+    @property
+    def manifest(self) -> PackageManifest:
+        manifest = PackageManifest(
+            sources=self.sources,
+            contractTypes=list(self.contracts.values()),
+            compilers=self.compiler_data,  # type: ignore
+        )
+
+        # NOTE: This property is optional here, but mandatory for publishing
+        if self.config.name:
+            manifest.name = self.config.name
+
+        # NOTE: This property is optional here, but mandatory for publishing
+        if self.config.version:
+            manifest.version = self.config.version
+
```

Comment:
```suggestion

        # NOTE: This property is optional here, but mandatory for publishing
        if self.config.meta:
            manifest.meta = self.config.meta
```

Reply:
I deleted this property because it wasn't being used and because the code was buggy (`sources=self.sources` was a `List[Path]` when it should have been a `Dict[str, Source]`). Want me to add it back?

I'm also confused by having both a `manifest` and `cached_manifest` properties. Which one do you use when? Does that `self.confg...` logic also need to apply to the `cached_manifest`? Right now `load_contracts` doesn't use `self.manifest` AFAICS.